src/app/utility/new-data-type-inline/new-data-type-inline.component.ts

```diff
@@ -70,6 +72,7 @@ export class NewDataTypeInlineComponent implements OnInit, AfterViewInit {
 
   ngOnInit() {
```

Comment:
I've identified a potential memory leak. This component does not have an `ngOnDestroy` function to clean up the subscription created in `ngAfterViewInit`:

```ts
this.formDataTypeChangesSubscription = this.myFormNewDataType.form.valueChanges.subscribe(
      x => {
        this.validate(x);
      }
    );
```

If this is not unsubscribed then the subscription is kept in memory. This should be cleaned up by:

1. Implementing the `OnDestroy` Angular interface on the component
2. Either:
    1. Calling `this.formDataTypeChangesSubscription.unubscribe()` in `ngOnDestroy`, or
    2. Use the declarative version of unsubscribe: https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe

Reply:
Yes, again, good spot.  I've gone with the first option - the declarative version looks cleaner, but doesn't seem to save much code, and I couldn't see it used elsewhere, so keeping things consistent.